scripts/version_buckets.py

```diff
@@ -0,0 +1,243 @@
+    account. Keys are bucket names. Only buckets related to the control panel
+    are returned.
+
+    Return only the `number` of buckets. If `number = 0` (the default), return
+    all available buckets.
+    """
+    result = {}
+    click.echo("Getting S3 bucket details.")
+    s3 = boto3.client("s3")
+    bucket_metadata = s3.list_buckets()
+    if number:
+        buckets = bucket_metadata["Buckets"][:number]
+    else:
+        buckets = bucket_metadata["Buckets"]
+    with click.progressbar(buckets, len(buckets)) as bucket_list:
+        for bucket in bucket_list:
+            name = bucket["Name"]
+            bucket_obj = boto3.resource("s3").Bucket(name)
+            if is_control_panel_bucket(name, bucket_obj):
+                versioning = bucket_obj.Versioning()
+                try:
+                    lifecycle_conf = bucket_obj.LifecycleConfiguration()
+                    lifecycle = lifecycle_conf.rules
+                except ClientError as ex:
+                    lifecycle = ""
+                result[name] = {
+                    "bucket": bucket_obj,
+                    "versioning": versioning.status,
+                    "lifecycle": lifecycle,
+                }
+    return result
+
+
+@click.group()
+@click.option("--verbose", is_flag=True, help="Comprehensive logging set to stdout.")
+def main(verbose=False):
+    """
+    A tool to manage the versioning and life cycle of S3 buckets.
+    """
+    if verbose:
+        global VERBOSE
+        VERBOSE = True
+        verbose_handler = logging.StreamHandler(sys.stdout)
+        verbose_handler.setLevel(logging.INFO)
+        verbose_handler.setFormatter(log_formatter)
+        logger.addHandler(verbose_handler)
+        click.echo("Logging to {}\n".format(LOGFILE))
+
+
+@main.command()
+@click.option("-n", "--number", required=False, type=int)
+def list(number=0):
+    """
+    List n (or all, if not given) bucket details.
+    """
+    buckets = get_buckets(number)
+    data = [("Name", "Versioned", "LifeCycle")]
+    col_width = [0, 0, 0]
+    rows = []
+    for name, bucket in buckets.items():
+        v = bucket["versioning"]
+        l = bucket["lifecycle"]
+        v = v if v else "Disabled"
+        if l:
+            l = json.dumps(l, indent=1)
+        else:
+            l = "None"
+        data.append((name, v, l))
+    for row in data:
+        for i, info in enumerate(row):
+            col_width[i] = min(max(len(info) + 2, col_width[i]), 48)
+    dashes = tuple(("-" * (width - 1) for width in col_width))
+    data.insert(1, dashes)
+    click.echo(f"The status of the buckets:")
+    for row in data:
+        output = ""
+        for i in range(3):
+            output += row[i].ljust(col_width[i])
+        if not VERBOSE:
+            click.echo(output)
+        logger.info(output)
+
+
+@main.command()
+@click.option("-n", "--number", required=False, type=int)
+@click.option("-x", "--execute", is_flag=True)
```

Comment:
Is `--execute` like a `--dummy`/`--dry-run` opposite? I saw some code commented below, I wonder if that's intentional.

If we comment code because it has side effect it may mean we may want to default to `False` if it worries us.

Reply:
I just wanted to be explicit that the command should have side effects so I didn't make changes without being explicit about it.

Reply:
We tend to call this `--dry-run` as it's the common name for flags like this and also to have it true by default so that it doesn't do things for real by default but I guess it doesn't matter ☺️

Reply:
Heh... depends on the tools I suppose. I've seen the `-x` (for `eXecute`) flag elsewhere. No big deal I agree... ;-)